### DISCUSSION TITLE
Limit number of pages displayed in pagination navigation

### DISCUSSION THREAD

**Discussion by adamgreenwell** [bootscore/bootscore#600] [Q&A] [answered]
I've copied the bootscore_pagination function to my child theme and have been trying to modify it to limit the number of pages in the list to 10. I've changed the $range variable to 9 which limits the items in the navigation list to 10 on the first page of the category where I am using the function. However, I have a lot of categories with a lot of pages and if I'm on page 21 for instance, I get 19 page options in the pagination navigation. I need to limit the number of items in the pagination navigation to 10 regardless of what page the user is currently on.

I've been looking at this for a while and just keep spinning wheels. I feel like this is a simple solution by editing this for loop

`for ($i = 1; $i <= $pages; $i ++) {
        if (1 != $pages && (!($i >= $paged + $range + 1 || $i <= $paged - $range - 1) || $pages <= $showitems)) {
          echo ($paged == $i) ? '<li class="page-item page-'. $i .' active"><span class="page-link"><span class="sr-only">' . __('Current Page', 'bootscore') . ' </span>' . $i . '</span></li>' : '<li class="page-item page-'. $i .'"><a class="page-link" href="' . get_pagenum_link($i) . '"><span class="sr-only">' . __('Page', 'bootscore') . ' </span>' . $i . '</a></li>';
        }
      }`

but I'm just not having any success. Can anyone point me in the right direction? I appreciate the help and apologize in advance if the solution is as obvious as I suspect it may be.

**Comment by crftwrk** [comment, marked as answer]
This won't work through the pagination, because pagination depends on the amount of posts displayed in the category. 

If you want pagination to only show 10 pages, you need to limit posts as described here https://wordpress.org/support/topic/how-to-limit-posts-and-pagination-pages-in-categories-tags/.

**Comment by adamgreenwell** [comment]
That's great information - thank you for your help! That gives me a lot to work with.

**Reply by crftwrk** [reply]
You‘re welcome👍

I reopen this discussion, because other users maybe have the same question.
